### PR TITLE
fix: increase coverage for `SuperVaults` [sup-9218]

### PR DESCRIPTION
**Pull request by TamaraRingas**
- Add test for non-management calling `createSuperVault()` in factory
- Add test for SuperformRegistry == zero address in factory
- Rm unused internal function in factory
- Rm duplicate check in SuperVault

### CODE REVIEW

File: src/SuperVault.sol [hunk 1]
```diff
@@ -129,10 +129,6 @@ contract SuperVault is BaseStrategy, ISuperVault {
             revert ZERO_ADDRESS();
         }
 
-        if (block.chainid > type(uint64).max) {
```
[the remark this answers is not in the document]
Comment (TamaraRingas) [reply src/SuperVault.sol:132]: I noticed we had two of this check, I think the other one should still be present on lines 139-141

**Review by 0xTimepunk — APPROVED** [review]
